exercises/complex-numbers/src/ComplexNumbers.hs

```diff
@@ -7,35 +7,39 @@ module ComplexNumbers
  mul,
  add,
  sub,
- div) where
+ div,
+ complex) where
 
 import Prelude hiding (div, abs)
 
 -- Data definition -------------------------------------------------------------
-data Complex = Dummy
+data Complex a = Dummy deriving(Eq, Show)
 
 -- unary operators -------------------------------------------------------------
-conjugate :: Complex -> Complex
+conjugate :: Num a => Complex a -> Complex a
 conjugate = error "You need to implement this function"
 
-abs :: Complex -> Float
+abs :: Floating a => Complex a -> a
 abs = error "You need to implement this function"
 
-real :: Complex -> Float
+real :: Num a => Complex a -> a
 real = error "You need to implement this function"
 
-imaginary :: Complex -> Float
+imaginary :: Num a => Complex a -> a
 imaginary = error "You need to implement this function"
 
 -- binary operators ------------------------------------------------------------
-mul :: Complex -> Complex -> Complex
+complex :: (a, a) -> Complex a
```

Comment:
I wouldn't put this in binary operators section. Binary operator implies `Complex a -> Complex a -> something`. This `complex` creates a complex number.

Reply:
Binary operator implies two arguments. But I see what you mean.